realm/realm-library/src/main/java/io/realm/internal/Util.java

```diff
@@ -96,46 +98,45 @@ public static boolean isEmptyString(@Nullable String str) {
         return str == null || str.length() == 0;
     }
 
+    /**
+     * To delete Realm and related temporary files. This must be called in
+     * {@link OsObjectStore#callWithLock(RealmConfiguration, Runnable)}'s callback.
+     *
+     * @return {@code true} if the realm file is deleted. Temporary file deletion failure will not impact on return
```

Comment:
on return -> the return

Reply:
Also, it will impact the return value right? `false` is returned if they cannot be deleted?

Reply:
no, it will always return true if the realm file has been deleted.

I think it is just good enough to ensure the realm file is deleted since they take most of disk spaces.

Also, for the Realm on external storage, the temporary files are not deleted since we don't know where are they. (the logic of this part is decided by core at runtiem :( where all the fifo permission nightmare come from)